bench/irmin-pack/trace_replay.ml

```diff
@@ -435,6 +427,7 @@ module Make (Store : Store) = struct
           in
           t.commits_since_start_or_gc <- t.commits_since_start_or_gc + 1;
           prog 1;
+          let* _ = Lwt.pause () in
```

Comment:
This is the line needed to give our async finalizer a chance to run (if you comment this out you will see that the "Gc ended..." line is never logged), but it breaks our progress bar for some reason.

Reply:
For the Tezos integration, it looks as though they already have purposefully placed calls to `Lwt.pause` to allow for this kind of scheduling cooperation.

https://gitlab.com/search?search=Lwt.pause&nav_source=navbar&project_id=3836952&group_id=1385864&search_code=true&repository_ref=master